dkpro-core-arktools-gpl/src/main/java/de/tudarmstadt/ukp/dkpro/core/arktools/ArktweetPosTaggerTrainer.java

```diff
@@ -0,0 +1,97 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+package de.tudarmstadt.ukp.dkpro.core.arktools;
+
+import static org.apache.uima.fit.util.JCasUtil.indexCovered;
+import static org.apache.uima.fit.util.JCasUtil.select;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.uima.analysis_engine.AnalysisEngineProcessException;
+import org.apache.uima.fit.component.JCasConsumer_ImplBase;
+import org.apache.uima.fit.descriptor.ConfigurationParameter;
+import org.apache.uima.fit.descriptor.MimeTypeCapability;
+import org.apache.uima.fit.descriptor.ResourceMetaData;
+import org.apache.uima.jcas.JCas;
+
+import cmu.arktweetnlp.Train;
+import de.tudarmstadt.ukp.dkpro.core.api.parameter.ComponentParameters;
+import de.tudarmstadt.ukp.dkpro.core.api.parameter.MimeTypes;
+import de.tudarmstadt.ukp.dkpro.core.api.segmentation.type.Sentence;
+import de.tudarmstadt.ukp.dkpro.core.api.segmentation.type.Token;
+
+@MimeTypeCapability(MimeTypes.APPLICATION_X_ARKTWEET_TAGGER)
+@ResourceMetaData(name = "ArkTweet POS-Tagger Trainer")
+public class ArktweetPosTaggerTrainer extends JCasConsumer_ImplBase {
+
+    public static final String PARAM_TARGET_LOCATION = ComponentParameters.PARAM_TARGET_LOCATION;
+    @ConfigurationParameter(name = PARAM_TARGET_LOCATION, mandatory = true)
+    private File targetLocation;
+
+    private File tempData;
+    private PrintWriter out;
+
+    @Override
+    public void process(JCas jCas) throws AnalysisEngineProcessException {
+        if (tempData == null) {
+            try {
+                tempData = File.createTempFile("dkpro-stanford-pos-trainer", ".tsv");
+                out = new PrintWriter(new OutputStreamWriter(new FileOutputStream(tempData),
+                        StandardCharsets.UTF_8));
+            }
+            catch (IOException e) {
+                throw new AnalysisEngineProcessException(e);
+            }
+        }
+
+
+        Map<Sentence, Collection<Token>> index = indexCovered(jCas, Sentence.class, Token.class);
+        for (Sentence sentence : select(jCas, Sentence.class)) {
+            Collection<Token> tokens = index.get(sentence);
+            for (Token token : tokens) {
+                out.printf("%s\t%s%n", token.getText(), token.getPos().getPosValue());
```

Comment:
If you have problems with locale-specific numbers, maybe explicitly add a locale to the printf here?

```
System.out.printf( Locale.ITALIAN, "The date is %tc\n", date );
```

Reply:
The arktweet tagger has a bug. I reported it back, see [here](https://github.com/brendano/ark-tweet-nlp/issues/45).